### PR TITLE
Improve summary views

### PR DESCRIPTION
**Pull request by thomasleese**
This makes a few changes to the "Check your answers" summary views including using localised strings where possible and fixing change links.

### CODE REVIEW

File: app/views/teacher_interface/personal_information/_summary.html.erb [hunk 1]
```diff
@@ -1,24 +1,28 @@
...
+      href: %i[name_and_date_of_birth teacher_interface application_form personal_information]
+    },
+    family_name: {
+      href: %i[name_and_date_of_birth teacher_interface application_form personal_information]
+    },
+    date_of_birth: {
+      href: %i[name_and_date_of_birth teacher_interface application_form personal_information]
+    },
+    has_alternative_name: {
+      key: "Name appears differently on your ID documents or qualifications?",
```
Comment (gpeng) [inline app/views/teacher_interface/personal_information/_summary.html.erb:15]:
```suggestion
      title: "Name appears differently on your ID documents or qualifications?",
```

File: app/views/teacher_interface/personal_information/_summary.html.erb [hunk 2]
```diff
@@ -1,24 +1,28 @@
...
+      href: %i[alternative_name teacher_interface application_form personal_information]
+    },
+    alternative_given_names: application_form.has_alternative_name? ? {
+      href: %i[alternative_name teacher_interface application_form personal_information]
+    } : nil,
+    alternative_family_name: application_form.has_alternative_name? ? {
+      href: %i[alternative_name teacher_interface application_form personal_information]
+    } : nil,
```
Comment (gpeng) [inline app/views/teacher_interface/personal_information/_summary.html.erb:23]: This must be why we handle `nil`. Seems a bit 🤮

**Review by gpeng — CHANGES_REQUESTED** [review]

**Comment by gpeng** [comment]
Looks good. Just needs a rebase for the key -> title change I merged this morning.

File: app/views/teacher_interface/personal_information/_summary.html.erb (hunk 1 above)
Comment (thomasleese) [reply app/views/teacher_interface/personal_information/_summary.html.erb]: I've amended the commit with these fixes!

File: app/views/teacher_interface/personal_information/_summary.html.erb (hunk 2 above)
Comment (thomasleese) [reply app/views/teacher_interface/personal_information/_summary.html.erb:23]: Ah yeah, it is a bit horrible, could be refactored I think.

**Review by gpeng — APPROVED** [review]

**Pull request merged**